content/posts/Tanmoy_Sarkar_Phase_1_Report.md

```diff
@@ -0,0 +1,141 @@
+
+
+### Integrate Ruby Debugger
+
+This integration enhances the debugging experience for developers by providing a visual debugging capability directly from their integrated development environment (IDE). It allows anyone to conveniently set breakpoints and debug the application from their IDE. This integration support  also docker based environment. By changing configuration in IDE , it can easily connect to TCP based debugger and debug the application easily. Documentation has been updated accordingly.
+
+##### References
+- Demo Video : [Click Here](https://github.com/CircuitVerse/CircuitVerse/assets/57363826/1c0e957e-3eed-4555-8670-27df282786ab)
+- PR : [Ruby debugger integration](https://github.com/CircuitVerse/CircuitVerse/pull/3760)
+
+
+### Integrate Solargraph LSP
+
+Solargraph is a Language Server Protocol (LSP) server that enhances autocompletion for Ruby codebases by providing YARD documentation. To further enhance support for Rails codebases, the solargraph-rails gem has been installed and configured. YARD documentation has been generated for the entire codebase to ensure seamless autocompletion.
+
+This integration also supports Docker-based environments, allowing the application to be run in a Docker container while connecting the IDE/Editor's solargraph extension to the Solargraph server via a TCP socket. Initially, there was a significant challenge in this setup. The extension used to send the current directory to the LSP server, but the LSP server had a different path inside the Docker container. However, this issue was successfully resolved by creating a symlink inside the container. 
+
+![solargraph-docker](/images/tanmoy_gsoc23/solargraph-docker.jpeg)
+
+For more detailed information, please refer to the blog post available at [https://tanmoy.online/week-1-2-circuitversegsoc23](https://tanmoy.online/week-1-2-circuitversegsoc23)
+
+Currently, documentation and support has been added for 
+
+* VS Code
+* Sublime Editor
+* Neovim
+
+##### References
+- Demo Video : [Click Here](https://github.com/CircuitVerse/CircuitVerse/assets/57363826/5ec17053-e7fe-4bfd-abb3-673b54711de1)
+- All PR's : [Integrate LSP with Solargraph](https://github.com/CircuitVerse/CircuitVerse/issues/3761)
+
+
+### Integrate Vite Rails
+
+Previously in the project all the javascript compiled in single one via esbuild and served by sprocket asset pipeline in development and production [mostly same]. As a result, during development on each change, esbuild need to build multiple JS files againg which increase friction during development. 
+
+That’s why vite rails integrated and simulator assets moved to vite. After this integration, if there are many changes in the javascript, only the spciefic portion will be updated i website and hot reload will be performed if possible, else complete page will reload. Also, instead of packing and serving javascript files, it will load all dependency of javascript files asynchronously in browser. 
```

Comment:
```suggestion
That’s why vite rails is integrated and simulator assets moved to vite. After this integration, if there are many changes in the javascript, only the specific portion will be updated in the website and hot reload will be performed (if possible), otherwise entire page will reload. Also, instead of packing and serving javascript files, it will load all dependency of javascript files asynchronously in browser. 
```